### ISSUE TITLE
（GAS）在庫管理システム海外用

### ISSUE THREAD

**Issue by ryuji0909** [ryuji0909/memo#1]
`function doGet() {
  const template = HtmlService.createTemplateFromFile('index');
  return template.evaluate();
}

function createTable(isFirstTable) {
  const id = "17KMA-kZ7PRLiHYR7_iOPwM2FVpxJ93w-YM0YZAIJuuk"; // スプレッドシートのIDを更新
  const ss = SpreadsheetApp.openById(id);
  const sheet = ss.getSheetByName("海外仕入先一覧"); // "海外仕入先一覧"シートを指定

  // 動的な範囲選択に変更。最終行と最終列を取得して範囲を指定します。
  const lastRow = sheet.getLastRow();
  const lastColumn = sheet.getLastColumn();
  const range = sheet.getRange(2, 1, lastRow - 1, lastColumn); // "A2"から最終行までの範囲を動的に指定
  const values = range.getValues();

  // ADD 2024/06/13 CODE順で並び替え
  values.sort(function(a, b) {
    if (a[0] < b[0]) return -1;
    if (a[0] > b[0]) return 1;
    return 0;
  });

  let result = "";
  let kana = "";

  // ADD 2024/06/13 falseの時だけヘッダーを表示
  if (!isFirstTable) {
    result += '<table>';
    result += '<thead>';
    result += '<tr>';
    result += '<th>CODE</th> <th>社名</th> <th>国</th> <th>通貨</th> <th>貿易条件</th> <th>支払条件</th><th>詳細ページ</th>';
    result += '</tr>';
    result += '</thead>';
    result += '</table>'; 
  }

  for (var i = 0; i < values.length; i++) {
    if (kana !== values[i][0]) {
      kana = values[i][0];
      result += '<table class="supplier-table" data-kana="' + kana + '">';
    }
    
    let row = values[i];
    
    // rowの値が空白だった場合"n/a"と表記
    for (var j = 0; j < row.length; j++){
      if(row[j] == ""){
        row[j] = "n/a";
      }
    }

    // 数字を4桁で表示、2桁や3桁の場合には先頭に0を追加
    var codes = row[0].toString().padStart(4, '0');

    result += '<td>' + codes + '</td> <td>' + row[1] + '</td> <td>' + row[2] + '</td> <td>' + row[21] + '</td> <td>' + row[22] + '</td> <td>' + row[24] + '</td>' + '<td><a href="https://script.google.com/a/macros/hq.stylinglife.co.jp/s/AKfycbxfFJFrdINADnpDCmwkLjTiX4iaVnI2t7l4f6npE5MYrVqES3LN_Ei6EydGPZhP5Xuj/exec?p2=' + encodeURIComponent(row[0]) + '"target="_blank">詳細ページ</a></td>';

    result += '</tr>';
  }
 
  result += '</table>';
  // console.log(result);
  return result;
}`

**Comment by ryuji0909** [comment]
`<!DOCTYPE html>
<html>
<head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
<style>
  .supplier-table {
    width: 100%;
    table-layout: fixed;
  }
  /*ADD 2024年06/10* 新規登録リンク*/
  .link a {
    display: inline-block;
    text-decoration: none;
    color: black;
    font-size: 16px;
  }
  .header {
    display: flex;
    align-items: center;
  }

  button {
    margin-left: 20px;
  }
  /*ADD 2024年06/10*　新規登録リンク*/

  table {
    width: 100%;
    border-collapse: separate;
    border-spacing: 0;
  }

  table th:first-child {
    border-radius: 5px 0 0 0;
  }

  table th:last-child {
    border-radius: 0 5px 0 0;
    border-right: 1px solid #3c6690;
  }

  table th {
    text-align: left;
    color: white;
    background: linear-gradient(#829ebc, #225588);
    border-left: 1px solid #3c6690;
    border-top: 1px solid #3c6690;
    border-bottom: 1px solid #3c6690;
    box-shadow: 0px 1px 1px rgba(255,255,255,0,3) inset;
    width: 10%;
    padding: 10px 10px;
  }

  table td {
    text-align: left;
    font-size: 15px;
    border-left: 1px solid #a8b7c5;
    border-bottom: 1px solid #a8b7c5;
    border-top: none;
    box-shadow: 0px -3px 5px 1px #eee inset;
    width: 10%;
    padding: 10px 0;
  }

  table td:last-child {
    border-right: 1px solid #a8b7c5;
  }

  table tr:last-child td:first-child {
    border-radius: 0 0 0 5px;
  }

  table tr:last-child td:last-child {
    border-radius: 0 0 5px 0;
  }

  .supplier-table th, .supplier-table td {
    padding: 10px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  h4 {
    color: red;
    font-size: 15px;
  }
</style>
</head>
<body>
  <div class="header">
    <h1>仕入先コード表</h1> 
    <div class="link">
      <button><a href="https://script.google.com/a/macros/hq.stylinglife.co.jp/s/AKfycbxfFJFrdINADnpDCmwkLjTiX4iaVnI2t7l4f6npE5MYrVqES3LN_Ei6EydGPZhP5Xuj/exec?p2=" target="_blank">新規登録</a></button>
    </div>
  </div>

  <h4>検索はCtrl ＋ Fで検索可能です。</h4>

  <div id="createdTable"></div>

<script>
  // ページが読み込まれた際にサーバーサイドからデータを取得しHTMLに表示
  function createInnerTable() {
    google.script.run
      .withSuccessHandler(function(result) {
          $('#createdTable').html(result);
      })
      .createTable();
  }
  window.addEventListener('load', createInnerTable);
</script>
</body>
</html>`
